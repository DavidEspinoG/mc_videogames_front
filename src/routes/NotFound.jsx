import '../styles/NotFound.scss';
import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

const NotFound = () => {
  const navigate = useNavigate();
  useEffect(() => {
    setTimeout(() => {
      navigate('/');
    }, 3000);
  });
  return (
    <div className="login-main-container">
      <div className="notfound-container">
        <h2 className="notfound-title">The page you&apos;re looking for does not exist</h2>
        <p className="redirecting">Redirecting to home..</p>
      </div>
    </div>
  );
};

export default NotFound;
