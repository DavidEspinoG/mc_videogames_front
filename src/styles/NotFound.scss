@keyframes redirecting {
  0% {
    color: white;
  }

  50% {
    color: var(--green);
  }

  100% {
    color: white;
  }
}

.notfound-container {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.notfound-title {
  color: white;
}

.redirecting {
  animation-name: redirecting;
  animation-duration: 1.5s;
  animation-iteration-count: infinite;
}
