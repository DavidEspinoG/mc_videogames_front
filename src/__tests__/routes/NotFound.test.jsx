import NotFound from '../../routes/NotFound';
import renderWithProviders from '../test-utils';

describe('NotFound', () => {
  it('should render as expected', () => {
    const { container, unmount } = renderWithProviders(<NotFound />);
    expect(container).toMatchSnapshot();
    unmount();
  });
});
