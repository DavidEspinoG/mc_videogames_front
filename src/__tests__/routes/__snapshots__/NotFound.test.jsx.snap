// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > should render as expected 1`] = `
<div>
  <div
    class="login-main-container"
  >
    <div
      class="notfound-container container"
    >
      <h2
        class="notfound-title"
      >
        The page you're looking for does not exist
      </h2>
      <p
        class="redirecting"
      >
        Redirecting to home..
      </p>
    </div>
  </div>
</div>
`;
